snuba/datasets/processors/search_issues_processor.py

```diff
@@ -110,6 +112,12 @@ def process_insert_v1(
         retention_days = enforce_retention(
             event.get("retention_days", 90), detection_timestamp
         )
+        client_timestamp = _ensure_valid_date(
+            datetime.strptime(event["datetime"], settings.PAYLOAD_DATETIME_FORMAT)
```

Comment:
Is it possible for the `datetime` field on the incoming event to be empty? If yes, then you should handle that case.

Reply:
Good point. I think we'll always have that present, but if its missing its probably better to fail the event processing instead of defaulting the value to received_timestamp. Re-worked the logic to reflect those changes.